### ISSUE TITLE
&nbsp; causes xml parser to fail

### ISSUE THREAD

**Issue by IvanToronto** [conorhastings/react-syntax-highlighter#127]
Once you place &nbsp; anywhere in your code snippet, the whole highlighting stops working.

**Comment by conorhastings** [comment]
can you post full code sample including use of SyntaxHighlighter

**Comment by IvanToronto** [comment]
<html>
<body>
some text &nbsp; another text
</body>
</html>


and the use is:
        <SyntaxHighlighter language='xml' wrapLines={true} style={vs} showLineNumbers={this.props.lineNumbers}>{this.props.code}
        </SyntaxHighlighter>

**Comment by conorhastings** [comment]
it looks fine to me -- https://codesandbox.io/s/zzq2jy1w3x -- can you tell me what the issue is there?

**Comment by IvanToronto** [comment]
The issue appears when I use slightly different imports at the top of the file
`import SyntaxHighlighter from 'react-syntax-highlighter/prism';
import { vs } from 'react-syntax-highlighter/styles/prism';`

With this configuration, try to add and remove & character and you will see that it affect the entire code highlighting

**Comment by IvanToronto** [comment]
Is there any way to achieve that selected color scheme, using hljs or I need to select only whatever is available in that color set?

**Comment by conorhastings** [comment]
you can write your own style object or use what's available from hljs, the styles don't translate between each other

**Comment by IvanToronto** [comment]
Ok, thanks

**Comment by wooorm** [comment]
What version of refractor is used? I updated it recently to fix entities in [2.5.0](https://github.com/wooorm/refractor/releases).

<details>
<summary>
And for me it works
</summary>

```js
var refractor = require('refractor')

var code = `<html>
<body>
  some text &nbsp; another text
</body>
</html>`

var nodes = refractor.highlight(code, 'xml')

console.dir(nodes, {depth: null})
```

Works:

```js
[ { type: 'element',
    tagName: 'span',
    properties: { className: [ 'token', 'tag' ] },
    children: 
     [ { type: 'element',
         tagName: 'span',
         properties: { className: [ 'token', 'tag' ] },
         children: 
          [ { type: 'element',
              tagName: 'span',
              properties: { className: [ 'token', 'punctuation' ] },
              children: [ { type: 'text', value: '<' } ] },
            { type: 'text', value: 'html' } ] },
       { type: 'element',
         tagName: 'span',
         properties: { className: [ 'token', 'punctuation' ] },
         children: [ { type: 'text', value: '>' } ] } ] },
  { type: 'text', value: '\n' },
  { type: 'element',
    tagName: 'span',
    properties: { className: [ 'token', 'tag' ] },
    children: 
     [ { type: 'element',
         tagName: 'span',
         properties: { className: [ 'token', 'tag' ] },
         children: 
          [ { type: 'element',
              tagName: 'span',
              properties: { className: [ 'token', 'punctuation' ] },
              children: [ { type: 'text', value: '<' } ] },
            { type: 'text', value: 'body' } ] },
       { type: 'element',
         tagName: 'span',
         properties: { className: [ 'token', 'punctuation' ] },
         children: [ { type: 'text', value: '>' } ] } ] },
  { type: 'text', value: '\n  some text ' },
  { type: 'element',
    tagName: 'span',
    properties: { className: [ 'token', 'entity' ], title: ' ' },
    children: [ { type: 'text', value: '&nbsp;' } ] },
  { type: 'text', value: ' another text\n' },
  { type: 'element',
    tagName: 'span',
    properties: { className: [ 'token', 'tag' ] },
    children: 
     [ { type: 'element',
         tagName: 'span',
         properties: { className: [ 'token', 'tag' ] },
         children: 
          [ { type: 'element',
              tagName: 'span',
              properties: { className: [ 'token', 'punctuation' ] },
              children: [ { type: 'text', value: '</' } ] },
            { type: 'text', value: 'body' } ] },
       { type: 'element',
         tagName: 'span',
         properties: { className: [ 'token', 'punctuation' ] },
         children: [ { type: 'text', value: '>' } ] } ] },
  { type: 'text', value: '\n' },
  { type: 'element',
    tagName: 'span',
    properties: { className: [ 'token', 'tag' ] },
    children: 
     [ { type: 'element',
         tagName: 'span',
         properties: { className: [ 'token', 'tag' ] },
         children: 
          [ { type: 'element',
              tagName: 'span',
              properties: { className: [ 'token', 'punctuation' ] },
              children: [ { type: 'text', value: '</' } ] },
            { type: 'text', value: 'html' } ] },
       { type: 'element',
         tagName: 'span',
         properties: { className: [ 'token', 'punctuation' ] },
         children: [ { type: 'text', value: '>' } ] } ] } ]
```

</details>



However, in the resulting tree you’ll not that `&nbsp;` is a child of the text node. That value is supposed to be visible in document, and not rendering as an actual no-break space. So maybe it could also go wrong at escaping that value?
